runtime/shared_common/CacheMap.cpp

```diff
@@ -375,11 +375,12 @@ SH_CacheMap::cleanup(J9VMThread* currentThread)
 	}
 	while (theCC) {
 		theCC->cleanup(currentThread);
-		theCC = theCC->getNext();
+		SH_CompositeCacheImpl* nextCC = theCC->getNext();
```

Comment:
Can this line be moved before the call to `cleanup()`?   Fetching the next before doing the cleanup ensures that future changes to cleanup don't mess this up

Reply:
Agreed and updated the fix as suggested above.